easybuild/easyblocks/c/code_saturne.py

```diff
@@ -0,0 +1,158 @@
+
+        self.log.info("Configuration step is running...")
+
+        # only use the opt flags
+        env.setvar("CFLAGS", os.environ['OPTFLAGS'])
+        env.setvar("CXXFLAGS", os.environ['OPTFLAGS'])
+        env.setvar("FCFLAGS", os.environ['OPTFLAGS'])
+
+        cmd = ' '.join([
+            './configure',
+            '--prefix=' + self.installdir,
+            '--without-modules',
+            self.cfg['configopts'],
+            ])
+
+        if self.cfg['debug']:
+            cmd = ' '.join([cmd, '--enable-debug'])
+
+        (out, _) = run_cmd(cmd, log_all=True, simple=False, log_output=True)
+
+        return out
+
+    def build_step(self):
+        """ Build step for Code_Saturne."""
+
+        paracmd = ''
+        if self.cfg['parallel']:
+            paracmd = "-j %s" % self.cfg['parallel']
+
+        cmd = "make %s" % paracmd
+
+        (out, _) = run_cmd(cmd, log_all=True, simple=False, log_output=True)
+
+        return out
+
+    def install_step(self):
+        """ Build step for Code_Saturne."""
+
+        cmd = "make install"
+
+        (out, _) = run_cmd(cmd, log_all=True, simple=False, log_output=True)
+
+        return out
+
+    def post_install_step(self):
+        """Custom post install step for Code_Saturne."""
+        super(EB_Code_underscore_Saturne, self).post_install_step()
+
+        # create a "etc/code_saturne.cfg" and modify it to match SLURM
+        if self.cfg['slurm']:
+
+            self.log.info("Running the post-install SLURM step ...")
+
+            target_path = os.path.join(self.installdir, 'etc/code_saturne.cfg')
+            from_path = target_path + '.template'
+
+            apply_regex_substitutions(from_path,
+                                      [(r"# batch =", r"batch = SLURM")])
+            apply_regex_substitutions(from_path,
+                                      [(r"# mpiexec = mpiexec", r"mpiexec = srun")])
+            apply_regex_substitutions(from_path,
+                                      [(r"# mpiexec_n = ' -n '", r"mpiexec_n = ' -n '")])
+            apply_regex_substitutions(from_path,
+                                      [(r"# mpiexec_n_per_node =", r"mpiexec_n_per_node = ' --ntasks-per-node '")])
+
+            os.rename(from_path, target_path)
+
+    def sanity_check_step(self):
+        """Custom sanity check step for Code_Saturne."""
+
+        shlib_ext = get_shared_lib_ext()
+
+        custom_paths = {
+            'files': ['bin/code_saturne', 'lib/libsaturne.%s' % shlib_ext],
+            'dirs': ['bin', 'lib', 'libexec', 'include', 'etc'],
+        }
+
+        super(EB_Code_underscore_Saturne, self).sanity_check_step(
+            custom_paths=custom_paths)
+
+    def make_module_extra(self, altroot=None, altversion=None):
+        """Extra environment variables for Code_Saturne."""
+
+        txt = super(EB_Code_underscore_Saturne, self).make_module_extra()
+
+        cs_bashPath = os.path.join(self.installdir, 'etc', 'bash_completion.d', 'code_saturne')
```

Comment:
I am not sure this variable name follows any particular code style. Change it maybe to `cs_bash_path`?

Reply:
sure, looks better.